library/spiders/books.py

```diff
@@ -0,0 +1,68 @@
+from typing import Dict
+from urllib.parse import urljoin
+
+
+import scrapy
+from scrapy.http import Response
+from twisted.internet.asyncioreactor import AsyncioSelectorReactor
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs) -> Dict:
+        for book in response.css(".product_pod"):
+            yield scrapy.Request(
+                url=urljoin(
+                    response.url,
+                    book.css("h3 > a::attr(href)").get()
+                ),
+                callback=self.get_single_book
+            )
+
+        next_page = response.css(".next > a::attr(href)").get()
+        if next_page is not None:
```

Comment:
is not None - not necessary

Reply:
yes I know I can use only if next_page, I think it is a good practice to check the None exactly like this. I will change this but mentor maybe can help in this topic

Reply:
No matter how you write this, but if functionality is the same then simpler == better?)